sunscreen_compiler_common/src/lookup.rs

```diff
@@ -101,7 +101,44 @@ impl Default for StackFrameLookup {
     }
 }
 
-type Group = String;
+// The first field is a group name, the second field is the source code associated with the group.
+// type Group = (String, String);
+/**
+ *
+ */
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct Group {
+    /** */
+    pub id: u64,
+    /** */
+    pub name: String,
+    /** */
+    pub source: String,
+    /** */
+    pub parent: u64,
```

Comment:
Make this `Option<u64>`, as the root doesn't have a parent. Since the root's node id is also zero, I suspect you're going to have bugs figuring out if you've hit the root or not.